index.rst

```diff
@@ -1,61 +1,294 @@
+
+     The DMS shall preserve and keep in an accessible state an alert archive with all issued alerts for a historical record and for false alert analysis.
+
+ - DMS-PRTL-REQ-0033 "Queries on the Alerts Database" :cite:`LDM-554`:
+
+     The Portal aspect shall provide a query interface to the Alert Database, allowing searches based on parameters which shall include, but may not be limited to, Alert ID, time of alert, position on the sky, filter, and alert characteristics.
+
+ - DMS-PRTL-REQ-0048 "Alert Visualization" :cite:`LDM-554`:
+
+     The Portal aspect shall provide for the users a "property sheet" for the contents of an alert packet including, but not necessarily limited to, the alert postage stamp image, the postage stamp time series, the photometric time series, the source and object information (e.g., position, brightness).
+
+Proposed Implementation
+=======================
+
+We can satisfy these design inputs by storing serialized Avro alert data (the same bytes sent via Kafka to brokers) in a S3-like object store, indexed by a unique alert ID.
+Each alert packet corresponds to one object in the object store.
 
 .. note::
 
-   **This technote is not yet published.**
+   An alternative would be to combine many packets into a block in the object store, perhaps of about 100 alert packets.
+   This might permit more efficient storage.
+   Storage might be more efficient because compression would be better when storing many alerts.
+   In informal experiments with simulated alert data, this requires about 5% less space to store than compressing each alert packet separately.
 
-   Design document for a database of all alerts sent by the Rubin observatory.
+   But this would be more complex, and make writing more difficult, as writes need to append to existing data which would require coordination between writers.
+   It would also make reading more complex; a separate index would need to be maintained which translates alert packet IDs into an identifier for the block containing the alert.
+   In light of these complexities, this design sticks to a simpler structure.
+
+An "object store" here is taken to mean an S3-like system.
+It should it scale well to handle many terabytes of data.
+The underlying object store should provide strong guarantees on data durability, either through redundancy or automated backups.
+It should also support write throughput which can keep up with the alert stream volume, handling at least 2,000 writes per second [1]_, with each write averaging around 40 kilobytes.
+Finally, it should support random read access with a median latency of under 2 seconds, permitting several hundred reads per second.
+
+Writes to the object store are handled by a Kafka consumer which copies alert packets from the main Kafka topic into the alert database.
+
+Reads are servered with a lightweight HTTP service and accompanying client library which allow retrieval by alert ID of any packet.
```

Comment:
Is "servered" spelled as you wanted?  I could see it either way!

Reply:
Oops, no, thanks.